lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -515,44 +512,43 @@ def join_app_task_ujs(self, exec_tasks, exec_apps, ujs_jobs):
             u_j_s['creation_time'] = u_j_s.pop('created')
             u_j_s['modification_time'] = u_j_s.pop('updated')
             u_j_s['estcompl'] = j.get('estcompl', None)
-            u_j_s['time_info'] = [u_j_s.get('creation_time'),
-                                  u_j_s.get('modification_time'),
-                                  u_j_s['estcompl']]
 
-            if u_j_s.get('authstrat', None) == 'kbaseworkspace':
-                u_j_s['wsid'] = u_j_s.get('authparam')
+            authparam = u_j_s.pop('authparam')
+            authstrat = u_j_s.pop('authstrat')
+            if authstrat == 'kbaseworkspace':
+                u_j_s['wsid'] = authparam
 
             if u_j_s.get('desc'):
-                desc = u_j_s['desc'].split()[-1]
+                desc = u_j_s.pop('desc').split()[-1]
                 if '.' in desc:
                     u_j_s['method'] = desc
 
             # Assuming complete, error and status all exist in the records returned
-            if not j.get('error'):
-                if j.get('complete'):
+            if not u_j_s.get('error'):
+                if u_j_s.get('complete'):
                     u_j_s['job_state'] = 'completed'
```

Comment:
why you want to keep job_state while it's not used anywhere and it's very expensive.

Reply:
Tian: this is a good question.  No one gave me a well defined required data items to start with, except the previous appcatalog/stats plus many inputs from the PD doc. And the main data source is called 'userjobstate', so it was natural to assume that is the major data wanted.  It was the main term used for this project.  And I had assumed it was required. When performance was not an issue, it was not a big deal.  

Until yesterday, when I went through the kbase-ui call, I realized the job_state from kb_Metrics was not used.  So I am fine to remove it.

However, as you might have noticed that Paramvir was requesting to have the client_groups displayed.  That was one of the original data items.